lib/saluki-io/src/deser/codec/dogstatsd/mod.rs

```diff
@@ -457,6 +472,82 @@ fn parse_dogstatsd_event<'a>(input: &'a [u8], config: &DogstatsdCodecConfigurati
     Ok((remaining, eventd))
 }
 
+fn parse_dogstatsd_service_check<'a>(
+    input: &'a [u8], config: &DogstatsdCodecConfiguration,
+) -> IResult<&'a [u8], ServiceCheck> {
+    let (remaining, (name, raw_check_status)) = preceded(
+        tag(message::SERVICE_CHECK_PREFIX),
+        separated_pair(ascii_alphanum_and_seps, tag(b"|"), parse_u8),
+    )(input)?;
+
+    let check_status = saluki_event::service_check::CheckStatus::try_from(raw_check_status)
```

Comment:
Nit: might just want to import `CheckStatus` directly to avoid needing the qualified path.